### PR TITLE
Added more information about context management to the OpenGL tutorial.

### PR DESCRIPTION
**Pull request by binary1248**
Title says everything.

### CODE REVIEW

**Comment by mantognini** [comment]
Added French translation.

File: tutorials/2.4/window-opengl-fr.php [hunk 1]
```diff
@@ -148,6 +151,90 @@
...
+    Chaque fenêtre créée dans SFML vient automatiquement avec un contexte
+    OpenGL. Quand une fonction OpenGL est appelée, cette dernière va opérer
+    sur le contexte actuellement actif. Il est donc requis qu'un contexte soit
+    actif à chaque fois qu'une fonction OpenGL est invoquée. Si aucun
+    contexte n'est activé à ce moment là, l'appel de fonction ne va pas
+    produire les effets escomptés car il n'y a pas d'état sur lequel la
+    fonction peut avoir un effet.
+</p>
+<p>
+    Pour faire en sort d'activer le contexte d'une fenêtre, utilisez
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: Pour faire en sort**e**

File: tutorials/2.4/window-opengl-fr.php [hunk 2]
```diff
@@ -148,6 +151,90 @@
...
+    <code>window.setActive()</code> qui est l'équivalent de
+    <code>window.setActive(true)</code>. Activer un contexte pendant qu'un
+    autre est actif a pour effet d'implicitement désactiver celui qui est
+    actuellement actif avant d'activer le nouveau contexte. Pour explicitement
+    désactiver le contexte d'une fenêtre, utiliez
+    <code>window.setActive(false)</code>. Comme expliqué plus loin, ceci est
+    requis si le contexte est à activer sur un autre <i>thread</i>. Cependant, de
+    manière générale il est recommandé de simplement désactivé un contexte à
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: de simplement désactiv**er**

File: tutorials/2.4/window-opengl-fr.php [hunk 3]
```diff
@@ -148,6 +151,90 @@
...
+    chaque fois que vous avez fini d'effectuer lot d'opérations OpenGL. En
+    suivant ce conseil, chaque groupe de telles opérations peut être entouré
+    entre des appels d'activation et de désactivation. Une classe d'utilitaire
+    <i>RAII</i> peut être conçus pour facilité cet effet.
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: conçu**e**

File: tutorials/2.4/window-opengl-fr.php [hunk 4]
```diff
@@ -148,6 +151,90 @@
...
+
+// configuration des états OpenGL
+// nettoyage des framebuffers
+// affichage dans la fenêtre
+
+// désactivation du contexte de la fenêtre
+window.setActive(false);
+</code></pre>
+<p class="important">
+    Pour déverminer des problèmes avec OpenGL et SFML, la première étape est
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: Sérieusement?? :D

Autant je suis attaché à une orthographe et un vocabulaire correct, "déverminer" je peux pas...

File: tutorials/2.4/window-opengl-fr.php [hunk 5]
```diff
@@ -148,6 +151,90 @@
...
+    toujours de vérifier qu'un contexte est activé lorsqu'une fonction OpenGL
+    est appelée. Ne partez pas du principe que SFML activera implicitement un
+    contexte ou que SFML va préserver le contexte actif lorsque vous invoquez
+    des fonctionnalités de la bibliothèque. La seule garantie est que le
+    contexte actif sur le <i>thread</i> courant ne changera pas entre les appels à
+    <code>window.setAct(true)</code> et <code>window.setActive(false)</code>
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: window.setAct**ive**(true)

File: tutorials/2.4/window-opengl-fr.php [hunk 6]
```diff
@@ -148,6 +151,90 @@
...
+    s'assurer que le même contexte soit de nouveau actif. De plus, il est
+    nécessaire que le bon contexte soit actif quand une fonction OpenGL est
+    appelée car les contextes servent à représenter l'environnement d'exécution
+    pour les opérations OpenGL ainsi que le <i>framebuffer</i> de destination
+    pour n'importe quelle commande de dessin. Il est à noter que, lorsqu'un
+    contexte sans <i>framebuffer</i> visible est actif, appeler une fonction
+    de dessin OpenGL ne produira aucun rendu visible. De même, répartir des
+    opérations OpenGL sur plusieurs contextes ne fera pas en sorte de propager
+    les changement d'états sur les différents contextes. Ainsi, si une
+    opération de dessin ultérieurs nécessite certains états, elle ne produira
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: ultérieur**e**

File: tutorials/2.4/window-opengl-fr.php [hunk 7]
```diff
@@ -148,6 +151,90 @@
...
+    pas les résultats escomptés.
+</p>
+<p class="important">
+    Lorsque vous écrivez du code OpenGL, il est fortement recommandé de
+    toujours vérifier si une erreur OpenGL a été produite après tout appel à
+    une fonction OpenGL. Ceci peut être accomplis avec la fonction
+    <code>glGetError()</code>. Vérifiez la présence d'erreur après chaque
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: Vérifie**r**

File: tutorials/2.4/window-opengl-fr.php [hunk 8]
```diff
@@ -148,6 +151,90 @@
...
+    invocation de fonction OpenGL vous aidera à cibler la source d'une possible erreur
+    et ainsi rendre le déverminage significativement plus efficace.
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: Autre occurence de "déverminage", si tu décides de le changer.

File: tutorials/2.4/window-opengl-fr.php [hunk 9]
```diff
@@ -148,6 +151,90 @@
...
+</p>
+<p>
+    Dépendamment de la version et des capacités du contexte disponible, il est
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: **En fonction** de la version

File: tutorials/2.4/window-opengl-fr.php [hunk 10]
```diff
@@ -148,6 +151,90 @@
...
+    important de n'appeler des fonctions qui soient compatibles avec ce
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: de n'appeler **que** des fonctions qui soient

File: tutorials/2.4/window-opengl-fr.php [hunk 11]
```diff
@@ -148,6 +151,90 @@
...
+    contexte. Ne pas respecter cette règle produira souvent des erreurs comme
+    <code>GL_INVALID_OPERATION</code> ou <code>GL_INVALID_ENUM</code>. Afin de
+    déterminer la version et capacité d'un contexte ayant été créé avec une
+    fenêtre ou séparément, <code>window.getSettings()</code> ou
+    <code>context.getSettings()</code> peuvent respectivement être utilisés.
+    Il se peut que les capacités et version d'un contexte diffèrent de celles
+    demandées lors de la création du contexte, notamment quand l'implémentation
+    d'OpenGL n'a pas été en mesure de répondre à toutes les exigences
+    demandées. Aussi, il est recommandé de toujours vérifier si le contexte
+    créé fournis bien les fonctionnalités requises pour exécuter le code
```
Comment (LaurentGomila) [inline tutorials/2.4/window-opengl-fr.php]: fourni**t**

**Comment by mantognini** [comment]
@LaurentGomila Thanks for proofreading this. I was expecting a few mistakes here and there, but not that many. :-S I'll fix those tomorrow or on Tuesday!

File: tutorials/2.4/window-opengl-fr.php (hunk 4 above)
Comment (mantognini) [inline tutorials/2.4/window-opengl-fr.php]: J'ai changé en déboguer/débogage. Pour ma défense, c'est comme ça que l'on me l'a enseigné. =P

**Comment by mantognini** [comment]
Ok, typos fixed -- I should really practice and write more often in French!

**Comment by Sonkun** [comment]
omg! sorry about fix-opengl-es branch, this is not where I meant to push
it....  (didnt know where to write this message either, so sorry for using
this thread....)

**Comment by Sonkun** [comment]
by the way, it contains interesting changes you might want to merge, but I
don't have time to do it (plus I know there are major ongoing changes in
the OpenGL backend), so feel free to reuse the changes/delete the branch

**Pull request merged**